app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -201,13 +203,23 @@ const Lease = ({ form, values, store }) => {
         </FormGrid.Row>
         <FormGrid.Row>
           <FormGrid.Item>
-            <SelectField
-              id='lease_assigned_unit'
-              label='Assigned Unit Number'
-              fieldName='lease.unit'
-              options={availableUnitsOptions}
-              disabled={disabled || areNoUnitsAvailable}
-              disabledOptions={areNoUnitsAvailable && noUnitsOptions}
+            <Field
+              name='lease.unit'
+              component={({ input: { onChange }, meta, ...rest }) => {
```

Comment:
nit: it doesn't look like you're using rest here, I think you can leave it out of the function params